src/pytorch_lightning/core/module.py

```diff
@@ -175,12 +179,13 @@ def lr_schedulers(self) -> Optional[Union[LRSchedulerTypeUnion, List[LRScheduler
 
     @property
     def trainer(self) -> "pl.Trainer":
-        if not self._running_torchscript and self._trainer is None:
+        if self._running_torchscript and self._trainer is not None:
+            return self._trainer
+        else:
             raise RuntimeError(f"{self.__class__.__qualname__} is not attached to a `Trainer`.")
-        return self._trainer
 
     @trainer.setter
-    def trainer(self, trainer: Optional["pl.Trainer"]) -> None:
+    def trainer(self, trainer: "pl.Trainer") -> None:
```

Comment:
see below the check whether trainer is None. 
cc @carmocca

Reply:
updated in https://github.com/Lightning-AI/lightning/pull/13603/commits/4ddf8415d562021c0bfd191bccda89d29d66d36f